commit 42c2bf99caadffc9a22e2839df4674e31f1f7712
Author: Kai Vehmanen <kvcontact@nosignal.fi>
Date:   Thu Jan 30 22:44:24 2014 +0200

    eca-osc: always initialize chainsetup edit objects
    
    Initialize 'need_chain_reinit' properly for OSC commands to modify
    chainop and controller parameters. Depending on stack contents, unnecessary
    chain reinits were performed.

diff --git a/libecasound/eca-osc.cpp b/libecasound/eca-osc.cpp
index 26040af..1cc675f 100644
--- a/libecasound/eca-osc.cpp
+++ b/libecasound/eca-osc.cpp
@@ -1,6 +1,6 @@
 // ------------------------------------------------------------------------
 // eca-osc.cpp: Class implementing the Ecasound OSC interface
-// Copyright (C) 2009,2012 Kai Vehmanen
+// Copyright (C) 2009,2012,2014 Kai Vehmanen
 //
 // Attributes:
 //     eca-style-version: 3
@@ -246,6 +246,7 @@ int ECA_OSC_INTERFACE::handle_chain_message(const std::string &path, const char
 	  p_res == 0) {
 	DBC_CHECK(types[0] == 'f');
 	edit.type = ECA::edit_cop_set_param;
+        edit.need_chain_reinit = false;
 
 	edit.m.cop_set_param.chain = c_index;
 	edit.m.cop_set_param.op = std::atoi(op_s.c_str());
@@ -277,7 +278,8 @@ int ECA_OSC_INTERFACE::handle_chain_message(const std::string &path, const char
 	  p_res == 0) {
 	DBC_CHECK(types[0] == 'f');
 	edit.type = ECA::edit_ctrl_set_param;
-
+        edit.need_chain_reinit = false;
+  
 	edit.m.ctrl_set_param.chain = c_index;
 	edit.m.ctrl_set_param.op = std::atoi(ctrl_s.c_str());
 	edit.m.ctrl_set_param.param = param;
